packages/sdk/src/evm/core/classes/contract-wrapper.ts

```diff
@@ -269,6 +269,94 @@ export class ContractWrapper<
     this.customOverrides = hook;
   }
 
+  /**
+   * @internal
+   */
+  public async prepare<
+    TMethod extends keyof TContract["functions"] = keyof TContract["functions"],
+  >(
+    functionName: TMethod,
+    ...args:
+      | Parameters<TContract["functions"][TMethod]>
+      | [...Parameters<TContract["functions"][TMethod]>, CallOverrides]
+  ) {
+    // parse last arg as tx options if present
+    let txOptions: CallOverrides | undefined;
+    try {
+      if (args.length > 0 && typeof args[args.length - 1] === "object") {
+        const last = args[args.length - 1];
+        txOptions = CallOverrideSchema.parse(last);
+        // if call overrides found, remove it from args array
+        args = args.slice(0, args.length - 1) as Parameters<
+          TContract["functions"][TMethod]
+        >;
+      }
+    } catch (e) {
+      // no-op
+    }
+
+    const functions = extractFunctionsFromAbi(AbiSchema.parse(this.abi)).filter(
+      (f) => f.name === functionName,
+    );
+
+    if (!functions.length) {
+      throw new Error(
+        `Function "${
+          functionName as string
+        }" not found in contract. Check your dashboard for the list of functions available`,
+      );
+    }
+    const fn = functions.find(
+      (f) => f.name === functionName && f.inputs.length === args.length,
+    );
+
+    // TODO extract this and re-use for deploy function to check constructor args
+    if (!fn) {
+      throw new Error(
+        `Function "${functionName as string}" requires ${
+          functions[0].inputs.length
+        } arguments, but ${
+          args.length
+        } were provided.\nExpected function signature: ${
+          functions[0].signature
+        }`,
+      );
+    }
+
+    const ethersFnName = `${functionName as string}(${fn.inputs
+      .map((i) => i.type)
+      .join()})` as TMethod;
+
+    // check if the function exists on the contract, otherwise use the name passed in
+    const fnName =
+      ethersFnName in this.readContract.functions ? ethersFnName : functionName;
+
+    // TODO validate each argument
+    if (fn.stateMutability === "view" || fn.stateMutability === "pure") {
+      throw new Error(
+        `Function ${
+          functionName as string
+        } is not a write function. Use call() instead.`,
+      );
+    } else {
+      const tx = await this.writeContract.populateTransaction[fnName as string](
+        ...(txOptions ? [...args, txOptions] : args),
+      );
+
+      const signer = this.getSigner();
+      if (!signer) {
+        throw new Error("No signer found");
+      }
+
+      const signedTx = await signer.signTransaction(tx);
+
+      return {
+        transaction: tx,
+        signature: signedTx,
```

Comment:
Hmmm i was hoping to get a transaction object here, something i can call estimate/simulate/encode on

Reply:
This setup is jake specific - I actually don't want to write like this I think - really what I want is for these to call into `Transaction`, but that causes a circular dependency which makes test cases fail for now so I just did the quick solution to get a working version for Jake

Also won't have duplicated `call` code